skops/io/tests/test_external.py

```diff
@@ -0,0 +1,325 @@
+    def lgbm(self):
+        lgbm = pytest.importorskip("lightgbm")
+        return lgbm
+
+    @pytest.fixture
+    def trusted(self):
+        # TODO: adjust once more types are trusted by default
+        return [
+            "collections.defaultdict",
+            "lightgbm.basic.Booster",
+            "lightgbm.sklearn.LGBMClassifier",
+            "lightgbm.sklearn.LGBMRegressor",
+            "lightgbm.sklearn.LGBMRanker",
+            "numpy.int32",
+            "numpy.int64",
+            "sklearn.preprocessing._label.LabelEncoder",
+        ]
+
+    boosting_types = ["gbdt", "dart", "goss", "rf"]
+
+    @pytest.mark.parametrize("boosting_type", boosting_types)
+    def test_classifier(self, lgbm, clf_data, trusted, boosting_type):
+        kw = {}
+        if boosting_type == "rf":
+            kw["bagging_fraction"] = 0.5
+            kw["bagging_freq"] = 2
+
+        estimator = lgbm.LGBMClassifier(boosting_type=boosting_type, **kw)
+        check_estimator(estimator, trusted=trusted)
+
+        X, y = clf_data
+        estimator.fit(X, y)
+        check_estimator(estimator, trusted=trusted)
+        check_method_outputs(estimator, X, trusted=trusted)
+
+    @pytest.mark.parametrize("boosting_type", boosting_types)
+    def test_regressor(self, lgbm, regr_data, trusted, boosting_type):
+        kw = {}
+        if boosting_type == "rf":
+            kw["bagging_fraction"] = 0.5
+            kw["bagging_freq"] = 2
+
+        estimator = lgbm.LGBMRegressor(boosting_type=boosting_type, **kw)
+        check_estimator(estimator, trusted=trusted)
+
+        X, y = regr_data
+        estimator.fit(X, y)
+        check_estimator(estimator, trusted=trusted)
+        check_method_outputs(estimator, X, trusted=trusted)
+
+    @pytest.mark.parametrize("boosting_type", boosting_types)
+    def test_ranker(self, lgbm, rank_data, trusted, boosting_type):
+        kw = {}
+        if boosting_type == "rf":
+            kw["bagging_fraction"] = 0.5
+            kw["bagging_freq"] = 2
+
+        estimator = lgbm.LGBMRanker(boosting_type=boosting_type, **kw)
+        check_estimator(estimator, trusted=trusted)
+
+        X, y, group = rank_data
+        estimator.fit(X, y, group=group)
+        check_estimator(estimator, trusted=trusted)
+        check_method_outputs(estimator, X, trusted=trusted)
+
+
+class TestXGBoost:
+    """Tests for XGBClassifier, XGBRegressor, XGBRFClassifier, XGBRFRegressor, XGBRanker
+
+    Known bugs:
+
+    - When initialzing with tree_method=None, its value resolves to "exact", but
+      after loading, it resolves to "auto" when calling get_params().
+    - When initializing with tree_method='gpu_hist' and gpu_id=None, the
+      latter's value resolves to 0, but after loading, it resolves to -1, when
+      calling get_params()
+
+    These discrepancies occur regardless of skops, so they're a problem in
+    xgboost itself. We assume that this has no practical consequences and thus
+    avoid testing these cases.
```

Comment:
is there an upstream issue for this on the xgboost side? If not, it'd be nice to open one and link to it here.

Reply:
As [I commented](https://github.com/skops-dev/skops/pull/244/#issuecomment-1344280205), I could open on issue but ideally someone who knows xgboost well could comment on whether it's really a bug or not. If there is no one around, I can just open no matter what.

Reply:
yeah I think the devs on the xgboost repo can decide if it's a bug which they wanna fix or not.

Reply:
https://github.com/dmlc/xgboost/issues/8596